zkstack_cli/crates/zkstack/src/commands/chain/gateway/migrate_to_gateway_calldata.rs

```diff
@@ -162,54 +175,57 @@ pub(crate) async fn get_migrate_to_gateway_calls(
         }
     };
 
-    let finalize_migrate_to_gateway_output = finalize_migrate_to_gateway(
-        shell,
-        forge_args,
-        foundry_contracts_path,
-        crate::admin_functions::AdminScriptMode::OnlySave,
-        params.l1_bridgehub_addr,
-        params.max_l1_gas_price,
-        params.l2_chain_id,
-        params.gateway_chain_id,
-        params.gateway_diamond_cut.into(),
-        refund_recipient,
-        params.l1_rpc_url.clone(),
-    )
-    .await?;
+    if !params.only_set_da_validator_pair {
+        let finalize_migrate_to_gateway_output = finalize_migrate_to_gateway(
+            shell,
+            forge_args,
+            foundry_contracts_path,
+            crate::admin_functions::AdminScriptMode::OnlySave,
+            params.l1_bridgehub_addr,
+            params.max_l1_gas_price,
+            params.l2_chain_id,
+            params.gateway_chain_id,
+            params.gateway_diamond_cut.into(),
+            refund_recipient,
+            params.l1_rpc_url.clone(),
+        )
+        .await?;
 
-    result.extend(finalize_migrate_to_gateway_output.calls);
+        result.extend(finalize_migrate_to_gateway_output.calls);
+    }
 
     // Changing L2 DA validator while migrating to gateway is not recommended; we allow changing only the settlement layer one
     let (_, l2_da_validator) = l1_zk_chain.get_da_validator_pair().await?;
+    if !l2_da_validator.is_zero() || params.only_set_da_validator_pair {
+        // Unfortunately, there is no getter for whether a chain is a permanent rollup, we have to
+        // read storage here.
+        let is_permanent_rollup_slot = l1_provider
+            .get_storage_at(zk_chain_l1_address, H256::from_low_u64_be(57), None)
+            .await?;
+        if is_permanent_rollup_slot == H256::from_low_u64_be(1) {
+            // TODO(EVM-1002): We should really check it on our own here, but it is hard with the current interfaces
+            logger::warn("WARNING: Your chain is a permanent rollup! Ensure that the new settlement layer DA provider is compatible with Gateway RollupDAManager!");
+        }
 
-    // Unfortunately, there is no getter for whether a chain is a permanent rollup, we have to
-    // read storage here.
-    let is_permanent_rollup_slot = l1_provider
-        .get_storage_at(zk_chain_l1_address, H256::from_low_u64_be(57), None)
+        let da_validator_encoding_result = set_da_validator_pair_via_gateway(
```

Comment:
Are you sure that including it in the script is the way to do it? It might make sense to have a separate command for it